eng/native/configureplatform.cmake

```diff
@@ -79,7 +79,11 @@ if(CLR_CMAKE_HOST_OS STREQUAL Darwin)
     set(CLR_CMAKE_HOST_UNIX 1)
     set(CLR_CMAKE_HOST_OSX 1)
     if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86_64)
-        set(CLR_CMAKE_HOST_UNIX_AMD64 1)
+        if(NOT CLR_CROSS_COMPONENTS_BUILD AND CLR_CMAKE_TARGET_ARCH STREQUAL arm64)
+            set(CLR_CMAKE_HOST_UNIX_ARM64 1)
+        else(CLR_CMAKE_TARGET_ARCH STREQUAL arm64)
+            set(CLR_CMAKE_HOST_UNIX_AMD64 1)
+        endif(CLR_CMAKE_TARGET_ARCH STREQUAL arm64)
     elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL arm64)
         set(CLR_CMAKE_HOST_UNIX_ARM64 1)
```

Comment:
When osx-arm64 becomes a primary dev platform, we will need similar code here when the target is x64.  Left this for future

Reply:
I ended up adding the other case too.